backend/src/main/java/com/htecgroup/skynest/service/impl/RefreshTokenServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.dto.UserDto;
+import com.htecgroup.skynest.model.jwtObject.JwtObject;
+import com.htecgroup.skynest.service.RefreshTokenService;
+import com.htecgroup.skynest.service.UserService;
+import com.htecgroup.skynest.util.JwtUtils;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class RefreshTokenServiceImpl implements RefreshTokenService {
+
+  private UserService userService;
+  private InvalidJwtServiceImpl invalidJwtService;
+
+  @Override
+  public String refreshToken(String refreshToken) {
+    if (refreshToken != null && refreshToken.startsWith(JwtUtils.TOKEN_PREFIX)) {
+      try {
+        String refresh_token = refreshToken.replace(JwtUtils.TOKEN_PREFIX, "");
+        String username = JwtUtils.getUsernameFromRefreshToken(refresh_token);
+        List<String> authorities =
+            JwtUtils.getAuthoritiesFromRefreshToken(refresh_token).stream()
```

Comment:
Since authorities can be changed from the time when refresh token was issued, it is better to read them from UserDto if you can istead